### PR TITLE
add sanbox command for crictl

### LINKED ISSUE #34: crictl sandbox
Add CLI for `crictl sandbox`, e.g.

```
crictl sandbox run config.json|yaml
crictl sandbox status sandbox-id
crictl sandbox stop sandbox-id
crictl sandbox remove sandbox-id
crictl sandbox list
```

### PR DESCRIPTION
**Pull request by heartlock**
close #34 

Signed-off-by: heartlock <21521209@zju.edu.cn>

### CODE REVIEW

File: cmd/crictl/sandbox.go [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+		},
+	},
+	Action: func(context *cli.Context) error {
+		// Set up a connection to the server.
+		conn, err := getRuntimeClientConnection(context)
+		if err != nil {
+			return fmt.Errorf("failed to connect: %v", err)
+		}
+		defer conn.Close()
+		client := pb.NewRuntimeServiceClient(conn)
```
Comment (xlgao-zju) [inline cmd/crictl/sandbox.go]: I think we can put these duplicated code in a new file, such as `util.go` or something you like.

File: cmd/crictl/sandbox.go [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+		if err != nil {
+			return fmt.Errorf("Stop podsandbox failed: %v", err)
+		}
+		return nil
+	},
+}
+
+// StopPodSandbox sends a StopPodSandboxRequest to server, and parses the returned StopPodSandboxRseponse
+func StopPodSandbox(client pb.RuntimeServiceClient, podsandboxID string) error {
+	_, err := client.StopPodSandbox(context.Background(), &pb.StopPodSandboxRequest{PodSandboxId: podsandboxID})
```
Comment (xlgao-zju) [inline cmd/crictl/sandbox.go]: Maybe we can print something here to tell users that the sandbox has been stopped.

File: cmd/crictl/sandbox.go [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+	if err != nil {
+		return err
+	}
+	return nil
```
Comment (xlgao-zju) [inline cmd/crictl/sandbox.go]: We can just return err here.

File: cmd/crictl/sandbox.go (hunk 2 above)
Comment (xlgao-zju) [inline cmd/crictl/sandbox.go]: Same as other places.

File: cmd/crictl/sandbox.go (hunk 3 above)
Comment (xlgao-zju) [inline cmd/crictl/sandbox.go]: Same as other places.

File: cmd/crictl/sandbox.go (hunk 1 above)
Comment (heartlock) [inline cmd/crictl/sandbox.go]: Good idea! I have a plan to do it.

File: cmd/crictl/sandbox.go (hunk 2 above)
Comment (heartlock) [inline cmd/crictl/sandbox.go]: nothing returned except err

Comment (xlgao-zju) [inline cmd/crictl/sandbox.go]: So, maybe [here](https://github.com/kubernetes-incubator/cri-tools/pull/40/files#diff-ef1517a1c727863c56e9fc00890cdd6cR89)?

File: cmd/crictl/sandbox.go [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+var runPodSandboxSpec string
+var podSandboxID string
+var filter string
+
+var runtimeRunPodSanboxCommand = cli.Command{
+	Name:  "runsandbox",
+	Usage: "run a sandbox",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:        "podsandbox-spec",
```
Comment (feiskyer) [inline cmd/crictl/sandbox.go]: The flag seems too long. Could we make it just an arg without flag, e.g. `crictl sandbox run config.json|yaml`?

File: cmd/crictl/sandbox.go [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+	if err != nil {
+		return fmt.Errorf("Parse podSandboxConfig failed: %v", err)
+	}
+
+	r, err := client.RunPodSandbox(context.Background(), &pb.RunPodSandboxRequest{Config: &podSandboxConfig})
+	if err != nil {
+		return err
+	}
+	fmt.Println("RunPodSandboxResponse:")
+	fmt.Println("podsandboxID: ", r.PodSandboxId)
```
Comment (feiskyer) [inline cmd/crictl/sandbox.go]: nit: merge two lines togather, e.g. "PodSandbox sandboxId created successfully"

File: cmd/crictl/sandbox.go [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+	return nil
+}
+
+var runtimeStopPodSanboxCommand = cli.Command{
+	Name:  "stoppodsandbox",
+	Usage: "stop a podsandbox",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:        "podsandbox",
```
Comment (feiskyer) [inline cmd/crictl/sandbox.go]: same here, it's better to use arg instead of flags (for more easier user-interface)

File: cmd/crictl/sandbox.go [hunk 7]
```diff
@@ -0,0 +1,225 @@
...
+	"os"
+
+	"github.com/urfave/cli"
+	"golang.org/x/net/context"
+	pb "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+)
+
+var runPodSandboxSpec string
+var podSandboxID string
+var filter string
```
Comment (feiskyer) [inline cmd/crictl/sandbox.go]: nit: define those variables where they are actually used.

File: cmd/crictl/sandbox.go [hunk 8]
```diff
@@ -0,0 +1,225 @@
...
+		return nil
+	},
+}
+
+// StopPodSandbox sends a StopPodSandboxRequest to server, and parses the returned StopPodSandboxRseponse
+func StopPodSandbox(client pb.RuntimeServiceClient, podsandboxID string) error {
+	_, err := client.StopPodSandbox(context.Background(), &pb.StopPodSandboxRequest{PodSandboxId: podsandboxID})
+	if err != nil {
+		return err
+	}
```
Comment (feiskyer) [inline cmd/crictl/sandbox.go]: Add a success output.

File: cmd/crictl/sandbox.go (hunk 4 above)
Comment (heartlock) [inline cmd/crictl/sandbox.go]: Sounds good! I will update it.

File: cmd/crictl/sandbox.go (hunk 8 above)
Comment (heartlock) [inline cmd/crictl/sandbox.go]: updated it! @feiskyer

File: cmd/crictl/sandbox.go [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+		stopPodSandboxCommand,
+		removePodSandboxCommand,
+		podSandboxStatusCommand,
+		listPodSandboxCommand,
+	},
+}
+
+var runPodSandboxSpec string
+
+//
```
Comment (feiskyer) [inline cmd/crictl/sandbox.go]: missing comments

File: cmd/crictl/sandbox.go [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+var runPodSandboxCommand = cli.Command{
+	Name:  "run",
+	Usage: "run a sandbox",
+	Action: func(context *cli.Context) error {
+		// Set up a connection to the server.
+		conn, err := getRuntimeClientConnection(context)
```
Comment (feiskyer) [inline cmd/crictl/sandbox.go]: it's better to check params first

File: cmd/crictl/util.go [hunk 11]
```diff
@@ -0,0 +1,31 @@
...
+		ls := make(map[string]string)
+		z := strings.SplitN(opt.Filter, "=", 2)
+		ls[z[0]] = z[1]
+		return &pb.PodSandboxFilter{
+			LabelSelector: ls,
+		}
+	}
+	return &pb.PodSandboxFilter{
+		State: &pb.PodSandboxStateValue{
+			State: pb.PodSandboxState_SANDBOX_READY,
```
Comment (feiskyer) [inline cmd/crictl/util.go]: also add state to SandboxListOptions?

Comment (heartlock) [inline cmd/crictl/util.go]: default shows just running sanbox?

Comment (feiskyer) [inline cmd/crictl/util.go]: No, if no filter is specified, it should show all sandboxes

**Comment by heartlock** [comment]
updated it! ready for review @feiskyer

File: cmd/crictl/sandbox.go [hunk 12]
```diff
@@ -0,0 +1,330 @@
...
+	"strings"
+	"time"
+
+	"github.com/urfave/cli"
+	"golang.org/x/net/context"
+	pb "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+)
+
+var runtimePodSandboxCommand = cli.Command{
+	Name: "sandbox",
```
Comment (feiskyer) [inline cmd/crictl/sandbox.go]: Please also a doc for "sandbox" command, e.g. set `Usage` here.

**Comment by feiskyer** [comment]
Help messages for subcommands are also missing:

```
# crictl stop -h
No help topic for 'stop'
# crictl run -h
No help topic for 'run'
```

**Comment by heartlock** [comment]
help messages for subcommands works:
```
# crictl sandbox stop -h
NAME:
   crictl sandbox stop - stop a pod sandbox

USAGE:
   crictl sandbox stop [arguments...]

# crictl sandbox run -h
NAME:
   crictl sandbox run - run a pod

USAGE:
   crictl sandbox run [arguments...]
```

**Comment by feiskyer** [comment]
Cool. LGTM.

**Pull request merged**